### DISCUSSION TITLE
Differences in generated images, between Comfy UI versions.

### DISCUSSION THREAD

**Discussion by JorgeR81** [Comfy-Org/ComfyUI#1981] [Q&A] [answered]
Hi,
I'm new to Comfy UI.

I have 2 Comfy UI portable installs downloaded on October 16 and November 14.
( I didn't use the update_comfy bat file.)

Both installs are working fine. I've got the new one, because the older one didn't work with some custom nodes I just downloaded.  

But I noticed the images generated are slightly different, between the 2 Comfy versions, even without using custom nodes.
The new images also look good, sometimes a bit better, other times not so mutch. 
All my prompts are fine tuned for the old Comfy install.

Was this visual change intentional, or an indirect consequence of other changes?
Does this happen often, during the Comfy UI development?
Is it possible to add an option, so that I can change between the 2 looks, without having 2 Comfy UI installs?

Thanks

![comfy1](https://github.com/comfyanonymous/ComfyUI/assets/72734038/be4f70c4-3938-4b72-97a4-64f27681ec67)
![comfy2](https://github.com/comfyanonymous/ComfyUI/assets/72734038/a2a044cb-080d-480f-a8ab-c8735fc68550)

**Comment by comfyanonymous** [comment]
There was a few issues fixed that are related to float imprecision and since this is a diffusion model sometimes tiny changes can make significant changes to the image especially if you use SDE samplers.

**Comment by JorgeR81** [comment]
Hi,
( first of all, I have to tell you how awesome is that you have a portable version ! )

For flat ( retro ) anime images ( dpmpp_2s_ancestral ( karras ), they are slightly different, but the quality seems to be about the same. 
But more realistic images ( dpmpp_2m ( karras ), they don't look as good to my eye. 
Across different seeds and checkpoints, with simple prompts, the new images have a slight green tint; they start to get blurry in places where it does not make sense for a photo, and the facial expression is just not as good.
I mean, they all still look good overall, but the new images are a bit worse, every single time. If it was a random thing it should be about 50/50, right ?

Did you had the chance to make image tests, after adding these changes? 
I don't know how extensive or necessary these changes were, but if possible I would suggest making a few tests and trying to implement them in a way that has less effects on the quality of the pictures.
Or maybe adding an option to generate images the old way, perhaps an extra option in the advanced sampling node? 


![comfy4](https://github.com/comfyanonymous/ComfyUI/assets/72734038/8bc31a8e-0141-46f4-a412-111885bc97cc)
![comfy3](https://github.com/comfyanonymous/ComfyUI/assets/72734038/f3079487-c130-4e1b-9e62-b50af9bc904e)

**Reply by ltdrdata** [reply]
If the model's precision becomes inaccurate, and as a result, the outcome appears better than normal, that is a side effect due to chance. It doesn't imply that it is correct or accurate.

Moreover, upon closer examination, the oct version shows a tendency for the anatomy around the clavicle area to deteriorate.

If, through broad testing across various models, it consistently produces good quality in a consistent manner, it could be considered for use as a single tuning method, akin to FreeU or Dynamic Thresholding. However, it cannot become a vanilla solution in itself.

**Reply by JorgeR81** [reply]
Yes, I also just noticed the clavicle, in the last picture, in the October version.
I was focusing more on the unrealistic blurring of the hair above, on that same side, in the November version. 

I tested over 50 realistic checkpoints ( edit: but only a few with the November version ) and sometimes the difference between checkpoints is less than the difference I'm getting between the 2 Comfy versions. 

This last picture is the epicrealism_pureEvolutionV5 checkpoint. 
That's a bit older ( from August ), but all the latest and best realistic models have some merge from epicRealism checkpoints. 
https://civitai.com/models/25694?modelVersionId=134065

Like the one in the picture above, RealLife 2.0. ( from November ) also with some epicRealism in the merge. But the clavicule issues are less noticeable ( and the seed is also different, by the way ).
https://civitai.com/models/171814/reallife

The worst November version in my tests so far, is the one in the first picture ( in my first post ). 
That's lomostyle, which I guess, is a checkpoint you would use when you want some sort of low quality effect. 
https://civitai.com/models/109923/lomostyle

Lastly, the second picture ( first post ) is animerge 2.1, to have a less realistic example. But also with some epicRealism model in the merge.
https://civitai.com/models/144249?modelVersionId=171496

No loras were used in the examples ( all checkpoints know how Mary Elizabeth Winstead looks, apparently! ) 
Prompts are nothing special. I used easynegative embedding for the negative prompt. 
 
If you are interested, I can provide many more examples.
 
Here is a new one from Consistent Factor (Euclid) 6.1 Cinematic, which is a checkpoint that takes much more creative liberties ( so she ends up looking more like Lea Seydoux! ).
https://civitai.com/models/9114?modelVersionId=142173

But on this one, I would say I like both versions about the same.
It would be nice to have both variations in the same install.  

![comfy5](https://github.com/comfyanonymous/ComfyUI/assets/72734038/4b76cbe4-2b8e-444c-a1e0-637bf1b9e556)

**Reply by comfyanonymous** [reply]
https://github.com/comfyanonymous/ComfyUI/commit/2a23ba0b8c225b59902423ef08db0de39d2ed7e7

https://github.com/comfyanonymous/ComfyUI/commit/111f1b525526a850cf222d2bccec0cdb3e2c988b

https://github.com/comfyanonymous/ComfyUI/commit/656c0b5d90239efb8be4281d2c16d52ca722064c (Specifically the "Don't apply weighting algorithm when weight is 1.0")

Those are specifically the commits that changed things, you can see they are just precision issue fixes so they shouldn't change quality at all.

**Comment by JorgeR81** [comment, marked as answer]
I just made the most embarrassing mistake !

I forgot I didn't have the easynegative embedding in the November version ! 
I added it and now both versions look identical ( like, pixel by pixel ! ).

I want you both to know that you're doing a great job, and I'm really sorry about wasting your time !
